### PR TITLE
add systemd input and lua filter support

### PR DESCRIPTION
**Pull request by wanjunlei**
Signed-off-by: wanjunlei <wanjunlei@yunify.com>

### CODE REVIEW

File: docs/user-guides/collect-systemd-logs.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Collect Service Logs of Kubernetes Nodes
+
+This tutorial guides you on how to collect Service logs of kubernetes nodes with fluentbit operator.
+
+# Demo
+
+First, install or update the fluentbit operator
```
Comment (benjaminhuo) [inline docs/user-guides/collect-systemd-logs.md:7]: Firstly

File: docs/user-guides/collect-systemd-logs.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+```shell
+kubectl apply -f manifests/setup/
+kubectl apply -f manifests/logging-stack/fluentbit-fluentBit.yaml
+kubectl apply -f manifests/logging-stack/fluentbitconfig-fluentBitConfig.yaml
+```
+
+Second, change the service logs directory
```
Comment (benjaminhuo) [inline docs/user-guides/collect-systemd-logs.md:15]: Secondly

File: docs/user-guides/collect-systemd-logs.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+```shell
+mkdir /var/log/journal/
+systemctl restart systemd-journald
+```
+
+Three, set up the fluentbit pipeline.
```
Comment (benjaminhuo) [inline docs/user-guides/collect-systemd-logs.md:22]: Thirdly

File: docs/user-guides/collect-systemd-logs.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+```shell
+kubectl apply -f manifests/setup/
+kubectl apply -f manifests/logging-stack/fluentbit-fluentBit.yaml
+kubectl apply -f manifests/logging-stack/fluentbitconfig-fluentBitConfig.yaml
+```
+
+Second, change the service logs directory
+
+```shell
+mkdir /var/log/journal/
```
Comment (benjaminhuo) [inline docs/user-guides/collect-systemd-logs.md:18]: Why do we need to recreate the journal directory?

File: manifests/logging-stack/input-systemd.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    tag: service.kubelet
+    path: /var/log/journal
+    db: /fluent-bit/tail/kubelet.db
+    dbSync: Normal
+    systemdFilter:
+    - _SYSTEMD_UNIT=kubelet.service
+
+---
+apiVersion: logging.kubesphere.io/v1alpha2
+kind: Input
```
Comment (benjaminhuo) [inline manifests/logging-stack/input-systemd.yaml:20]: docker is going to be replaced by other container runtimes like containerd or cri-o, so I think we can have 2 seperated inputs here which are input-kubelet.yaml and input-docker.yaml . Then user or installer will have an option to not installing the docker one

File: manifests/logging-stack/output-systemd-elasticsearch.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+apiVersion: logging.kubesphere.io/v1alpha2
+kind: Output
+metadata:
+  name: systemd
+  namespace: kubesphere-logging-system
+  labels:
+    logging.kubesphere.io/enabled: "true"
+    logging.kubesphere.io/component: logging
+spec:
+  match: service.*
```
Comment (benjaminhuo) [inline manifests/logging-stack/output-systemd-elasticsearch.yaml:10]: Is it possible to use one single output for each output type by setting `match: kube.*|service.*` in the following file？It'll be convenient to modify one single output in UI

https://github.com/kubesphere/fluentbit-operator/blob/master/manifests/logging-stack/output-elasticsearch.yaml
https://github.com/kubesphere/fluentbit-operator/blob/master/manifests/logging-stack/output-forward.yaml
https://github.com/kubesphere/fluentbit-operator/blob/master/manifests/logging-stack/output-kafka.yaml

File: manifests/setup/fluentbit-operator-deployment.yaml [hunk 7]
```diff
@@ -39,7 +39,7 @@ spec:
           mountPath: /var/run/docker.sock
       containers:
       - name: fluentbit-operator
-        image: 'kubespheredev/fluentbit-operator:v0.3.0'
+        image: 'kubesphere/fluentbit-operator:v0.3.0'
```
Comment (benjaminhuo) [inline manifests/setup/fluentbit-operator-deployment.yaml:42]: I think maybe we can upgrade the version to v0.4.0

**Review by benjaminhuo — CHANGES_REQUESTED** [review]

File: docs/user-guides/collect-systemd-logs.md (hunk 4 above)
Comment (wanjunlei) [reply docs/user-guides/collect-systemd-logs.md:18]: The default directory is /run/log/, if use the default directory, the fluent bit can not keep track of Journald cursor.

File: manifests/setup/fluentbit-operator-deployment.yaml (hunk 7 above)
Comment (wanjunlei) [reply manifests/setup/fluentbit-operator-deployment.yaml:42]: Ok, i will release a new version after this pr is megerd and then change this.

File: manifests/logging-stack/output-elasticsearch.yaml [hunk 8]
```diff
@@ -7,7 +7,7 @@ metadata:
     logging.kubesphere.io/enabled: "true"
     logging.kubesphere.io/component: logging
 spec:
-  match: kube.*
+  matchRegex: (?:kube|service)\.(.*)
```
Comment (benjaminhuo) [inline manifests/logging-stack/output-elasticsearch.yaml:10]: I think all the 3 outputs should make similar changes:
image

File: docs/user-guides/collect-systemd-logs.md (hunk 4 above)
Comment (benjaminhuo) [reply docs/user-guides/collect-systemd-logs.md:18]: in our dev env, there is already an existing `/var/log/journal` , so maybe we can describe this like `Please create directory /var/log/journal if it doesn't exist:`
image

**Review by benjaminhuo — CHANGES_REQUESTED** [review]

**Review by benjaminhuo — APPROVED** [review]

**Pull request merged**